File: server/api/analytics.py
import os

import segment.analytics as analytics
from prisma import enums

analytics.write_key = os.getenv("EXPO_PUBLIC_SEGMENT_WRITE_KEY")

def track_sent_message(
    user_id: str,
    chat_id: str,
    chat_type: str
):
    if os.getenv("EXPO_PUBLIC_SEGMENT_WRITE_KEY") is None:
        return

    analytics.track(
        user_id=user_id,
        event="Chat Message Sent",
        properties={
            "chat_id": chat_id,
            "chat_type": chat_type
        },
    )

File: lib/analytics.ts
import { createClient } from '@segment/analytics-react-native';

export const segmentClient = createClient({
  writeKey: process.env.EXPO_PUBLIC_SEGMENT_WRITE_KEY!
});

export const segmentTrackLoadedAuthPage = async () => {
  await segmentClient.track('Loaded Auth Page')
}

export const segmentTrackOpened = async (userId: string) => {
  await segmentClient.identify(userId)
  await segmentClient.track('Opened')
}

export const segmentTrackSignedIn = async (userId: string) => {
  await segmentClient.identify(userId)
  await segmentClient.track('Signed In')
}

export const segmentTrackFinishedSignup = async () => {
  await segmentClient.track('Finished Signup')
}

export const segmentTrackStartChat = async () => {
  await segmentClient.track('Start Chat')
}

export const segmentTrackEndChat = async () => {
  await segmentClient.track('End Chat')
}